generic/README.md

```diff
@@ -0,0 +1,157 @@
+```
+piranha-tree-sitter 0.1.0
+
+USAGE:
+    piranha-tree-sitter [OPTIONS] --path-to-codebase <PATH_TO_CODEBASE> --language <LANGUAGE> --flag-name <FLAG_NAME> --flag-namespace <FLAG_NAMESPACE> --path-to-configuration <PATH_TO_CONFIGURATION>
+
+OPTIONS:
+    -c, --path-to-configuration <PATH_TO_CONFIGURATION>
+            Folder containing the required configuration files
+    -f, --flag-name <FLAG_NAME>
+            Name of the stale flag
+    -h, --help
+            Print help information
+    -l, --language <LANGUAGE>
+    -n, --flag-namespace <FLAG_NAMESPACE>
+    -p, --path-to-codebase <PATH_TO_CODEBASE>
+            Input source code folder
+    -t, --flag-value
+            is treated?
+    -V, --version
+            Print version information
+```
+
+## Configuring Piranha  
+### Adding a flag API and specifying how it will be updated
+Consider the following definition of a rule in the `input_rules.toml` file. 
+```
+[[rules]]
+name = "Enum Based, toggle enabled"
+query = """((
+	(method_invocation 
+    	name : (_) @n1
+        arguments: ((argument_list 
+        				([
+                          (field_access field: (_)@f)
+                          (_) @f
+                         ])) )
+        	
+    ) @mi
+)
+(#eq? @n1 "isToggleEnabled")
+(#eq? @f "@stale_flag_name")
+)"""
+replace_node = "mi"
+replace = "@treated"
+groups = [ "Replace with boolean"]
+holes = ["treated", "stale_flag_name"]
+```
+This specifies a rule that matches against expressions like `exp.isToggleEnabled(SOME_FLAG_NAME)` and replaces it with `true` or `false`. 
+The `rule` contains `holes` or template variables that need to be instantiated.
+For instance, in the above rule `@treated` and `@stale_flag_name` need to be replaced with some concrete value so that the rule matches only the feature flag API usages corresponding to a specific flag, and replace it specifically with `true` or `false`. 
+The `query` property of the rule contains a tree-sitter query that is matched against the source code. 
+The node captured by the tag-name specifed in the `replace_node` property is replaced with the pattern specified in the `replace` property.
+The `replace` pattern can use the tags from the `query` to construct a replacement based on the match (like regex-replace).
+Each rule also contains the `groups` property, that specifies the kind of change performed by this rule. Based on this group, appropriate 
+cleanup rules will be performed by Piranha. For instance, `Replace with boolean literal` will trigger deep cleanups (like eliminating `consequent` of a `if statement`) to eliminate dead code caused by replacing an expression with a boolean literal. 
+Currently, Piranha provides deeper clean up after edits that `Replace with boolean literal` , `Delete Statement`, and `Delete Method`.
+
+### Configuring a new language 
+At a higher level, Piranha updates the feature flag API usages related to the stale flag and then cleans up the resulting dead code. 
```

Comment:
Let us add a note that this section can be skipped for most Piranha users unless they are planning on adding support for a new language.